enumerables.rb

```diff
@@ -0,0 +1,121 @@
+module Enumerable
+    def my_each
+        if block_given?
+            item = 0
+            while item < self.size
+                yield(self[item])
+             item+=1
+            end
+        end
+    
+    self
+    end
+
+
+
+    def my_each_with_index
+        if block_given?
+            index = 0
+            while index < self.size
+                yield(self[index], index)
+                index+=1
+            end
+        end
+        self
+    end
+
+    def my_select
+        if block_given?
+            filtered = []
+            index = 0
+            self.my_each { |x| filtered.push(x) if yield(x) }
+        end
+        filtered
+    end
+    
+    def my_all?
+        if block_given?
+            not_all = 0
+            index = 0
+            while index < self.size
+                not_all = 1 if !yield(self[index])
+                index+=1
+                break if not_all == 1
+            end
+            
+        end
+        not_all == 1 ? false : true
+    end
+
+    def my_any?
+        if block_given?
+            any = 0
+            index = 0
+            while index < self.size
+                any = 1 if yield(self[index])
+                index+=1
+                break if any == 1
+            end
+        end
+        any == 1 ? true : false
+    end
+
+    def my_none?
+        if block_given?
+            some = 0
+            index = 0
+            while index < self.size
+                some = 1 if yield(self[index])
+                index+=1
+                break if some == 1
+            end
+        end
+        some == 1 ? false : true
+    end
+
+    def my_count
+        if block_given?
+            count = 0
+            index = 0
+            while index < self.size
+                count+=1 if yield(self[index])
+                index+=1
+            end
+        end
+        count
+    end
+
+    def my_map(proc = nil)
+        new_array = []
+        index = 0
+        if proc 
+            self.my_each { |x| new_array << proc.call(x)}
+        end
+        if block_given?
+            self.my_each {|x| new_array << yield(x)}
+        end             
+        new_array
+    end
+
+    def my_inject
+        if block_given?
+            acumulator = self[0]
+            index = 1
+            while index < self.size
+                acumulator = yield(acumulator, self[index])
+                index+=1
+            end
+
+        end
+        acumulator
+    end
```

Comment:
`my_inject` should be able to do the following:
- accepts a symbol with or without an initial value.

   > If you specify a symbol instead, then each element in the collection will be passed to the named method of _memo_
   ```ruby
   # a sym with no initail value
   [1,2,3].inject(:+) #=> 6
   [1,2,3].my_inject(:+) #=> ArgumentError (wrong number of arguments (given 1, expected 0))
   
   # a sym with an initial value
   [1,2,3].inject(10,:+) #=> 16
   [1,2,3].my_inject(10,:+) #=> ArgumentError (wrong number of arguments (given 2, expected 0))
   ```
- when a block is given, it can also accept an initail value as an arugument
   >  If you specify a block, then for each element in enum the block is passed an accumulator value (memo) and the element. 
   ```ruby
   [1,2,3].inject(10){|a,b| a * b} #=> 60
   [1,2,3].my_inject(10){|a,b| a * b} #=> ArgumentError (wrong number of arguments (given 1, expected 0))
   ```

#### Please [check `inject` ruby doc for more details](https://ruby-doc.org/core-2.6.5/Object.html#method-i-to_enum). 

HINT: Investigate ruby's [`send`](https://apidock.com/ruby/Object/send) method, as `2 + 3 == 2.send(:+, 3)`. Here, `:+` is the symbol and operator.